.github/workflows/test-code.yml

```diff
@@ -76,6 +76,11 @@ jobs:
         if: ${{ startsWith(matrix.os, 'windows') }}
         working-directory: src
         run: dotnet test --no-restore --no-build
+      - name: Test AOT compilation
+        working-directory: src/AotSampleApp
+        run: |
+          dotnet publish -c Release -o publish
+          publish/AotSampleApp
```

Comment:
This will fail the build check if the application (a) returns a non-zero exit code, or (b) throws an exception which returns a non-zero exit code.